cmd/swarmctl/network/common.go

```diff
@@ -32,6 +32,12 @@ func GetNetwork(ctx context.Context, c api.ControlClient, input string) (*api.Ne
 		}
 
 		if l := len(rl.Networks); l > 1 {
+			for _, n := range rl.Networks {
+				if n.Spec.Annotations.Name == input {
+					// Found a full name match
```

Comment:
When you say "name", you mean "hostname", right? It's definitely possible for nodes to have the same hostname, and that's outside our control.